refinery/analysis_manager/tests.py

```diff
@@ -0,0 +1,398 @@
+
+import mock
+from django.contrib.auth.models import User
+from django.http import HttpResponseBadRequest, HttpResponseNotAllowed
+from django.test import RequestFactory, TestCase
+from guardian.utils import get_anonymous_user
+
+from analysis_manager.tasks import run_analysis
+from analysis_manager.utils import (create_analysis,
+                                    fetch_objects_required_for_analysis,
+                                    validate_analysis_config)
+from analysis_manager.views import run
+from core.models import Analysis, Workflow, WorkflowEngine
+from factory_boy.utils import (create_dataset_with_necessary_models,
+                               make_analyses_with_single_dataset)
+from galaxy_connector.models import Instance
+
+
+class AnalysisConfigTests(TestCase):
+    """
+    Test for validation of the two legacy ways of launching Analyses (
+    using NodeSets and NodeRelationships) as well as for the new Tool-based
+    approach
+    """
+    def test_valid_nodeset_analysis_config(self):
+        validate_analysis_config(
+            {
+                "custom_name": "Valid NodeSet Analysis Config",
+                "studyUuid": str(uuid.uuid4()),
+                "nodeSetUuid": str(uuid.uuid4()),
+                "user_id": 1,
+                "workflowUuid": str(uuid.uuid4())
+            }
+        )
+
+    def test_invalid_nodeset_analysis_config_missing_field(self):
+        with self.assertRaises(RuntimeError):
+            validate_analysis_config(
+                {
+                    "custom_name": "Missing a field",
+                    "studyUuid": str(uuid.uuid4()),
+                    "nodeSetUuid": str(uuid.uuid4()),
+                    "workflowUuid": str(uuid.uuid4())
+                }
+            )
+
+    def test_invalid_nodeset_analysis_config_non_uuid_field(self):
+        with self.assertRaises(RuntimeError):
+            validate_analysis_config(
+                {
+                    "custom_name": "Non-uuid field",
+                    "studyUuid": str(uuid.uuid4()),
+                    "nodeSetUuid": str(uuid.uuid4()),
+                    "user_id": 1,
+                    "workflowUuid": "Coffee"
+                }
+            )
+
+    def test_invalid_nodeset_analysis_config_non_int_field(self):
+        with self.assertRaises(RuntimeError):
+            validate_analysis_config(
+                {
+                    "custom_name": "Non int field",
+                    "studyUuid": str(uuid.uuid4()),
+                    "nodeSetUuid": str(uuid.uuid4()),
+                    "user_id": str(uuid.uuid4()),
+                    "workflowUuid": str(uuid.uuid4())
+                }
+            )
+
+    def test_valid_noderelationship_analysis_config(self):
+        validate_analysis_config(
+            {
+                "custom_name": "Valid NodeRelationship Analysis Config",
+                "studyUuid": str(uuid.uuid4()),
+                "nodeSetUuid": str(uuid.uuid4()),
```

Comment:
It doesn't seem like the name of the test is in sync with what's actually being tested? Perhaps consider a util method to construct your test cases, so the thing you're tweaking will stand out?